### PR TITLE
Adds Service Jackets to SO and MP vendors

### PR DESCRIPTION
**Pull request by Johannes2262**
Staff officer vendors are now stocked with a service jacket

# About the pull request

Staff officers have the Service Jacket of their role inside their vendors.

# Explain why it's good for the game

Staff Officers can now wear a Service Jacket if they believe, it is more fashionable without having to wait for a whitelisted CO to open the uniform vendors.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: Johannes2262
add: Service jackets in SO vendors
/:cl:

### CODE REVIEW

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Not MPs, feel free to add to doctor/CMO though.

**Comment by Johannes2262** [comment]
I didn't know there was a special doctor service jacket, and the normal one seemed to not fit to the other doctor clothing, I will look for it. About the MP's, the service jacket inside the vendors was meant for MPs helping out CIC or so, but you're right about that it would be abused for faster running speed with two extra slots.

**Comment by stanalbatross** [comment]
"More fashionable" ≠ good and these will inevitably be powergamed to hell and back. Service jackets should ONLY be in the locked uniform vendors as that was the point of them being moved there in the first place

**Comment by Johannes2262** [comment]
I removed the MP service jackets now.

File: code/game/machinery/vending/vendor_types/crew/staff_officer.dm [hunk 1]
```diff
@@ -15,6 +15,7 @@ GLOBAL_LIST_INIT(cm_vending_clothing_staff_officer, list(
 		list("Headset", 0, /obj/item/device/radio/headset/almayer/mcom, MARINE_CAN_BUY_EAR, VENDOR_ITEM_MANDATORY),
 		list("Helmet", 0, /obj/item/clothing/head/helmet/marine/MP/SO, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_MANDATORY),
 		list("MRE", 0, /obj/item/storage/box/MRE, MARINE_CAN_BUY_MRE, VENDOR_ITEM_MANDATORY),
+		list("Service Jacket", 0, /obj/item/clothing/suit/storage/jacket/marine/service/, MARINE_CAN_BUY_MRE, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/crew/staff_officer.dm:18]: The service jacket buy would overlap with MRE buyability. MARINE_CAN_BUY_MRE needs changed to one of the other defines that fits.

**Review by morrowwolf — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/crew/staff_officer.dm (hunk 1 above)
Comment (Johannes2262) [reply code/game/machinery/vending/vendor_types/crew/staff_officer.dm]: The service jacket and MRE buyability overlapping were fixed, the service jacket was changed to MARINE_CAN_BUY_ESSENTIALS.

**Pull request closed without merging**